core/app/models/concerns/spree/calculated_adjustments.rb

```diff
@@ -18,7 +18,7 @@ def calculator_type
 
       def calculator_type=(calculator_type)
         klass = calculator_type.constantize if calculator_type
-        self.calculator = klass.new if klass && !calculator.is_a?(klass)
+        self.calculator = klass.new if klass && !calculator.instance_of?(klass)
```

Comment:
@mrbrdo can you add some tests for this behavior?

Reply:
At the moment I am too busy with getting my store up and running. But I did include an example of what is the problem in the description of the PR. Subclassing a calculator in your own project won't work correctly without this fix.

Reply:
Would you be so kind to either add the test you had in mind or merge it, since all tests are still passing?